tests/testthat/test-model-summary.R

```diff
@@ -183,4 +183,27 @@ withr::with_options(list(bbr.bbi_exe_path = read_bbi_path()), {
 
     expect_error(model_summary(mod2), regexp = "More than one `\\.lst` file")
   })
+
+  test_that("model_summary() maps objective function fallback to NA [BBR-SUM-010]", {
+    skip_if_old_bbi("3.1.1")
+    withr::with_tempdir({
+      fs::dir_copy(MOD1_ABS_PATH, "tmpmod")
+      mod <- copy_model_from(read_model(MOD1_ABS_PATH),
+                             file.path(getwd(), "tmpmod"))
+
+      lst_file <- file.path("tmpmod", "1.lst")
+      lst_lines <- readr::read_lines(lst_file)
+      stringr::str_replace(lst_lines,
+                           "^( OBJECTIVE FUNCTION VALUE .*: +)([0-9.]+)\\s*$",
+                           "\\1NaN")  %>%
+        stringr::str_replace("^( #OBJV:\\*+ +)([0-9.]+)( +\\*+\\s*)$",
+                             "\\1NaN\\1")  %>%
```

Comment:
This happens to work, but the second group in the replacement should be `\\3`, not `\\1`.  And, like above, there's no reason to capture the numeric value.